scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -344,7 +344,7 @@ def handle(self, line):
         if self.status == TwisterStatus.PASS:
             tc.status = TwisterStatus.PASS
         else:
-            tc.status = TwisterStatus.FAIL
+            tc.status = self.status
```

Comment:
@golowanow, the case can be skipped, not FAIL.

Reply:
the reason to set to FAIL is described here:
https://github.com/zephyrproject-rtos/zephyr/blob/67a433fa048705444a46e1d1871dca6c89e41f75/scripts/pylib/twister/twisterlib/harness.py#L319-L325

Reply:
I think it is correct to set TC status to FAIL here either as a default status for any console line received while there is no clear PASS yet, or translating any harness non-Pass to TC failure. Assigning from harness.status might not harm as well (needs testing), but it is mostly just to keep it with same None here.

> Transition from passed to passed is not allowed ```, it is because the FAIL status is wrongly set by default.

IMO in practice it is ok to allow both PASS->PASS, or set FAIL by default.